assets/src/edit-story/utils/getCaretCharacterOffsetWithin.js

```diff
@@ -44,6 +44,9 @@ function getCaretCharacterOffsetWithin(element, clientX, clientY) {
     if (clientX && clientY) {
       if (doc.caretPositionFromPoint) {
         range = document.caretPositionFromPoint(clientX, clientY);
+        if (range && range.offset) {
+          return range.offset;
```

Comment:
This unfortunately doesn't work. There's a reason this function is as complicated, as it is. We need the character offset from the start of the full text container, not just the start of the node that the clicked text is inside.

Try making part of the text bold/italic/underline, defocus and the click inside the bold/italic/underline text. You'll see the offset registered as the offset of the nearest-node only: clicking between the 2nd and 3rd letter of a bold word will register as "offset=2" for the entire container, and then open the text editor with focus between the 2nd and 3rd letter of the entire text field overall. Which is very likely wrong.

Demo of current branch where editor consistently opens at the wrong offset in Firefox:
![click-offset](https://user-images.githubusercontent.com/637548/78716395-d5a1ff00-78ec-11ea-99f3-ce78161efffe.gif)

What the original code did was: create a new selection all the way from the start of the container up to and including the same offset in the same node as discovered from the click, and the get the length of that selection in characters (across all selected nodes).

If this doesn't work in Firefox, we need to figure out a way to do the same thing in Firefox at least.

Reply:
Eh, thanks for the information about the formatted text, didn't test with that!
Will look into it again.

Reply:
@barklund Should be improved now, PTAL.